controller/impl.go

```diff
@@ -32,16 +34,22 @@ func (f *Factory) startController(ctx context.Context, tenantID, policyID, polic
 	cleanup := func() {
 		stop <- true
 	}
-
+	i := 0
 	go func() {
 		for {
+			if i > maxRetries {
+				logger.Info().Msg("command loop reached maximum command loop retries, stopping")
+				stop <- true
+				return
+			}
 			err = f.runCommandLoop(ctx, &logger, policyID, policyName, instanceLabel, host, commandFunc, stop, options)
 			if err == nil || err == io.EOF {
 				return
 			}
 
 			logger.Info().Err(err).Msg("command loop exited with error, restarting")
 			time.Sleep(5 * time.Second)
```

Comment:
It would be better to use exponential backoff instead of a fixed timeout. 
Is there any knowledge we can derive from the returned error to decide what to do?